q2_types/feature_table/_transformer.py

```diff
@@ -72,6 +74,29 @@ def _dataframe_to_table(df):
                       sample_ids=df.index)
 
 
+def _biom_to_taxonomy_format(table):
+    axis = 'observation'
+    metadata = table.metadata(axis=axis)
+    if metadata is None:
+        raise TypeError('Table must have metadata')
+
+    taxonomy = []
+    for m in metadata:
+        if 'taxonomy' not in m.keys():
+            raise KeyError('Observation does not contain `taxonomy` metadata')
+
+        taxa = m['taxonomy']
+        if taxa is None:
```

Comment:
remove None check, wrap `append` call in try/except (note expected and observed types in error msg)